### PR TITLE
release: 2.61.2

### PR DESCRIPTION
**Pull request by ernestl**
Cherry picked all outstanding milestone 2.61 PRs (following commit fcf5729d361a266abfc3085bd241160c3df9fa53):
- [overlord: fix prerequisite base snap silently not being installed due to conflict](https://github.com/snapcore/snapd/pull/13589)
- [c/snap: fix prepare-image vs seedwriter wrt plugs/slots sanitization](https://github.com/snapcore/snapd/pull/13462)
- [o/devicestate: fix panic when requesting a serial when device-service.access=offline](https://github.com/snapcore/snapd/pull/13466)
- [o/devicestate: remodel policy/checks changes](https://github.com/snapcore/snapd/pull/13480)
- [c/snap: validate-seed also needs proper plugs/slots sanitizing](https://github.com/snapcore/snapd/pull/13472)
- [i/builtin: allow to lock /run/netns with network-control](https://github.com/snapcore/snapd/pull/13500)
- [naming: add well-known core24 snap-id](https://github.com/snapcore/snapd/pull/13531)
- [many: remove stub services for prompting](https://github.com/snapcore/snapd/pull/13536)
- [interfaces/builtin: Add ProductIDPattern of GoTrust Idem Key with NFC and usb-c](https://github.com/snapcore/snapd/pull/13389)
- [snapd.apparmor.service: add explicit dependency to snapd.mounts.target](https://github.com/snapcore/snapd/pull/13570)
- [data/systemd: make snapd.seed wait for snapd.socket only](https://github.com/snapcore/snapd/pull/13099)
- [overlord: restore services if the unlink fails](https://github.com/snapcore/snapd/pull/13582)
- [cmd/snap-confine: add libcudnn.so to list of NVIDIA libraries](https://github.com/snapcore/snapd/pull/13504)

Commits [o/devicestate: adjust test for Remodel signature](https://github.com/snapcore/snapd/pull/13600/commits/67d00690036bab108bed7d557b3bf72a5530cdbf) to [o/devicestate: pass empty slice in TestRemodelCannotUseOldModel](https://github.com/snapcore/snapd/pull/13600/commits/ae8697520c76420d573c5d8ea1bb5810d7b22e98) was added to fix compatibility issues in tests that resulted in static test failures. This was not done as backports, to save time.

Picked test fixes:
- [tests: remove opensuse 15 4](https://github.com/snapcore/snapd/pull/13444)
- [interfaces/apparmor: update apparmor template of s-u-n for changes in Go](https://github.com/snapcore/snapd/pull/13399)
- [t/m/snap-quota-memory: improve robustness of test on some distros](https://github.com/snapcore/snapd/pull/13399)

Labelled all cherry-picked PRs and generated changelogs with:
`DEBEMAIL="Ernest Lotter <ernest.lotter@canonical.com>" release-tools/changelog.py 2.61.2 2039017 NEWS.md`

NEWS.md and changelogs was updated in the final commit: [release: 2.61.2](https://github.com/snapcore/snapd/pull/13600/commits/8b2024be3c7e7d0099bcfbf71b18f1191c3da95d)

### CODE REVIEW

**Comment by Meulengracht** [comment]
It seems we are missing something here:
```
et: overlord/managers_test.go:4560:27: NewSequenceFromSnapSideInfos not declared by package snapstatetest
# github.com/snapcore/snapd/overlord/devicestate_test
vet: overlord/devicestate/devicestate_remodel_test.go:303:56: not enough arguments in call to devicestate.Remodel
	have (*state.State, *asserts.Model, nil, nil)
	want (*state.State, *asserts.Model, []*snap.SideInfo, []string, devicestate.RemodelOptions)
# github.com/snapcore/snapd/overlord/snapstate_test
vet: overlord/snapstate/handlers_link_test.go:663:27: NewSequenceFromSnapSideInfos not declared by package snapstatetest
```

**Comment by pedronis** [comment]
It's using the definitions as on master after the changes Alfonso did for
components. It needs to be changed to respect Sequence being just a
[]*SideInfo. The other seems independent things having landed related to
remodel, Remodel has new signature.

File: NEWS.md [hunk 1]
```diff
@@ -1,3 +1,21 @@
+# New in snapd 2.61.2:
+* Fix to enable plug/slot sanitization for prepare-image
+* Fix panic when device-service.access=offline
+* Support offline remodeling
+* Allow offline update only remodels without serial
+* Fail early when remodeling to old model version
+* Fix to enable plug/slot sanitization for validate-seed
+* Allow removal of core snap on classic systems
```
[the remark this answers is not in the document]
Comment (ernestl) [reply NEWS.md:8]: A backport was created for this: https://github.com/snapcore/snapd/pull/13595

File: NEWS.md [hunk 2]
```diff
@@ -1,3 +1,21 @@
+# New in snapd 2.61.2:
+* Fix to enable plug/slot sanitization for prepare-image
+* Fix panic when device-service.access=offline
+* Support offline remodeling
+* Allow offline update only remodels without serial
+* Fail early when remodeling to old model version
```
[the remark this answers is not in the document]
Comment (ernestl) [reply NEWS.md:6]: fixed

**Review by pedronis — APPROVED** [review]
thanks

**Comment by ernestl** [comment]
Failures Analysis:
-------------------------
(1) spread (ubuntu-22.04-64)                                              <- Still happens on master, ok to ignore.   

- google:ubuntu-22.04-64:tests/main/i18n

(2) spread (ubuntu-23.10-64)                                              <- No minimal test fix available, ok to ignore.

-  google:ubuntu-23.10-64:tests/main/lxd-mount-units
-  google:ubuntu-23.10-64:tests/main/lxd:snapd_cgroup_just_outside
-  google:ubuntu-23.10-64:tests/main/lxd-postrm-purge
-  etc...
--
lxc launch --quiet images:ubuntu/mantic my-ubuntu
Error: Failed instance creation: Failed getting remote image info: Failed getting image: The requested image couldn't be found for fingerprint "ubuntu/mantic"
--
images: does not hold images anymore. https://github.com/snapcore/snapd/pull/13426 addresses this by referencing an updated remote, but does a lot more to enable core24, so there is no minimal fix for 2.61.

(3) spread (ubuntu-core-22-64)                                           <- Still happens on master, ok to ignore.
- google:ubuntu-core-22-64:tests/main/interfaces-many-core-provided

Recommended fixes for this does not work: https://github.com/snapcore/snapd/pull/13515, https://github.com/snapcore/snapd/pull/13561 and problem still happens on master. 

(4) spread (ubuntu-core-22-arm-64)                                    <- Still happens on master, ok to ignore
- Error debugging google-arm:ubuntu-core-22-arm-64:tests/main/ (feb170753-706979) : EOF

(5) spread-nested (ubuntu-16.04-64)                                  <- ???
- google-nested:ubuntu-16.04-64:tests/nested/manual/devmode-snaps-can-run-other-snaps
--
Failed to run 'override-pull': Exit code was 1.
--
(6) spread-nested (ubuntu-20.04-64)                                  <- ???
- google-nested:ubuntu-20.04-64:tests/nested/manual/remodel-offline:local_snaps
- google-nested:ubuntu-20.04-64:tests/nested/manual/remodel-offline:local_assertions
- google-nested:ubuntu-20.04-64:tests/nested/manual/remodel-offline:installed_snaps
- google-nested:ubuntu-20.04-64:tests/nested/manual/remodel-offline:local_and_installed_snaps
- google-nested:ubuntu-20.04-64:tests/nested/manual/install-min-size

(7) spread-nested (ubuntu-22.04-64)                                  <- ???
- google-nested:ubuntu-22.04-64:tests/nested/manual/fde-on-classic
- google-nested:ubuntu-22.04-64:tests/nested/manual/muinstaller-real:plain
- google-nested:ubuntu-22.04-64:tests/nested/manual/muinstaller-real:plain
- google-nested:ubuntu-22.04-64:tests/nested/manual/install-min-size

**Pull request merged**